### ISSUE TITLE
 resource gcore_instancev2 with metadata_map containing key with special character ends up in an error 

### ISSUE THREAD

**Issue by x-killa** [G-Core/terraform-provider-gcore#153]
I have the following resource:
```
resource "gcore_instancev2" "mongodb" {
  flavor_id     = "g1-standard-2-4"
  name = "some_name"
  metadata_map = {
    "db.responsible" = "dpt"
  }
...
}
```
Terraform can successfully create the instance but fails next time with any operation with the following error:
```
Planning failed. Terraform encountered an error while generating this plan.
Error: cannot get metadata with key: {instance_id}. Error: Resource not found
```
The issue here is that provider is using v1 endpoint to get metadata value by key which indeed doesn't support special characters and returns 404 which results in the above mentioned error: https://api.gcore.com/cloud/v1/instances/{project_id}/{region_id}/{instance_id}/metadata/{key}

Seems like gcore_instancev2 should use v2 endpoint for metadata operations:
https://api.gcore.com/cloud/v2/instances/{project_id}/{region_id}/{instance_id}/metadata_item

**Comment by IsabellaMerkulova** [comment]
That's right, thank you so much for pointing out this bug to us. We will fix it soon. 
Alternatively, you can use keys without special characters for now.
We'll keep you updated!
